src/toil_scripts/lib/programs.py

```diff
@@ -83,36 +112,73 @@ def docker_call(tool,
     if env:
         for e, v in env.iteritems():
             base_docker_call.extend(['-e', '{}={}'.format(e, v)])
+    
     if docker_parameters:
         base_docker_call += docker_parameters
+   
+    docker_call = []
+    
+    run_pipe = False
 
-    _log.debug("Calling docker with %s." % " ".join(base_docker_call + [tool] + parameters))
-
-    docker_call = base_docker_call + [tool] + parameters
+    if bool(tools) == bool(tool):
+        raise Exception('Either "tool" or "tools" must contain a value, but not both.')
+    if not tools:
+        tools = [ tool ]
+    else:
+        run_pipe = True
+ 
+    # Pipe functionality
+    #   each element in the parameters list must represent a sub-pipe command
+    shell_flag = True      # Flag for running subprocess with string command or list command
+    if run_pipe:
+        command_list = []
+        if len(tools) > 1:
+            # If tool is a list containing multiple docker container name strings
+            #   then format the docker call in the 'pipe-of-containers' mode
+            docker_call.extend(base_docker_call + ['--entrypoint /bin/bash', tools[0], '-c \'{}\''.format(parameters[0])])
+            for i in xrange(1, len(tools)):
+                docker_call.extend(['|'] + base_docker_call + ['-i --entrypoint /bin/bash', tools[i], '-c \'{}\''.format(parameters[i])])
+            docker_call = " ".join(docker_call)
+            _log.debug("Calling docker with %s." % docker_call)
+            
+        elif len(tools) == 1:
+            # If tool is a list containing a single docker container name string
+            #   then format the docker call in the 'pipe-in-single-container' mode
+            docker_call.extend(base_docker_call + ['--entrypoint /bin/bash', tools[0], '-c \'{}\''.format(" | ".join(parameters))])
+            docker_call = " ".join(docker_call)
+            _log.debug("Calling docker with %s." % docker_call)
+             
+    else:        
+        docker_call = " ".join(base_docker_call + tools + parameters)
+        _log.debug("Calling docker with %s." % docker_call)
 
+    
     try:
         if outfile:
-            subprocess.check_call(docker_call, stdout=outfile)
+            subprocess.check_call(docker_call, stdout=outfile, shell=shell_flag)
+        elif check_output and return_stderr:
+            return subprocess.check_output(docker_call, shell=shell_flag, stderr=subprocess.STDOUT)
+        elif check_output:
+            return subprocess.check_output(docker_call, shell=shell_flag)
+        elif return_stderr:
+            return subprocess.check_call(docker_call, stderr=subprocess.STDOUT, shell=shell_flag)
```

Comment:
This isn't right. The user is passing `return_stderr=True` but they are getting the return value of check_call which is None. This also shows me that there are no tests for this case.